### ISSUE TITLE
Proposal: Maintain a list of pages needing translations

### ISSUE THREAD

**Issue by Keating950** [tldr-pages/tldr#4020]
I'd like to help translate some pages into French, but it's not obvious which pages are missing. Maintaining a list of pages that need translation into different languages might help lower the barrier to entry. It's also something that could be automated fairly easily by comparing directory contents. I would be happy to write a quick shell script to do it if that would be helpful.

**Comment by sbrl** [comment]
Hey, thanks for opening an issue about this @Keating950! We've actually been discussing this already in #3591. The core of the problem is that while we'd like to use an online translation service to manage translations, we haven't yet been able to find one that suits our needs.

I'm suspecting that we're going to have to develop a custom solution.

If you'd be able to write a quick shell script, that'd be great! It would be really cool to see what you've got in mind. We don't have anything at the moment, so thanks for the offer!

**Comment by Keating950** [comment]
Thanks for the reply! I took it a step further and wrote a quick [Python script](https://github.com/Keating950/tldr/commit/9132f7de27b5ec8d46099d025d49665123bb961b) to generate a markdown file,  but I'm not sure how best it would fit into the project's workflow. A pre-commit or push git hook seems like the obvious solution, but filling out the template requires jinja2, which some users might not have. What do you think would work best?

**Comment by sbrl** [comment]
Ah nice @Keating950!

Yeah, that's a great point. I think that the best place for this to run actually would be in the Travis CI build, since we can control the environment there and it avoids issues when using the GitHub web interface (?).

The problem though is we need to decide how to display the information. How do we let people know that the translated version of a page needs updating?

Currently, Travis CI communicates with [tldr-bot](https://github.com/tldr-pages/tldr-bot) - a bot that leaves comments on PRs if linting errors are found.

Perhaps with this we can finally start to crack the managing translations issue. Though I will warn you that the process of actually merging your contribution is likely to be a long, drawn-out, and complicated one - but thanks so much for stepping up and getting the ball rolling on this! :+1:

/cc @waldyrious, @agnivade, @mebeim, @arunisaac

**Comment by Keating950** [comment]
Happy to help! If what I've written turns out to be useful in the end, that's great, and if it doesn't work out that's fine too. In terms of pages missing translations, a markdown file in the main repo would be the first place I would look as a user, but the wiki seems like another possibility to me. I'll think about a possible solution for pages needing updates -- I can definitely see the complexity there. Let me know if there's anything I can do!

**Comment by sbrl** [comment]
Thanks again @Keating950!

The key issue with pages needing updates is not only identifying which ones are out of date, but also notifying the right people who would be able to update them. Perhaps we could identify 3-5 of the most recent contributors to a given language and @mention them in a comment or something? I'm not sure.

**Comment by Keating950** [comment]
I've [put something together](https://github.com/Keating950/tldr/commit/64d4650b80cbb92f9a8a782a15d59c82d5a91409) in my fork that should integrate with Travis. Would it be easier to keep track of at this point if I opened a PR? 

re: #3591, while these scripts would definitely need modification to integrate with a  web front-end, the basic diffing logic should be reusable, I think. I've never developed for that context before, though, so I'm speculating somewhat.

**Comment by sbrl** [comment]
Yeah, if you don't mind a PR would be great!

Ah, ok. We don't necessarily have to go with a web front-end, but that certainly a good way of doing it.

**Comment by agnivade** [comment]
I think dynamically updating a wiki page for every commit that went into master is a great idea.

**Comment by sbrl** [comment]
That's just the thing @agnivade! That's a great idea.

**Comment by bl-ue** [comment]
Is this what we're looking for, here: https://lukwebsforge.github.io/tldri18n/?

**Comment by sbrl** [comment]
It is! Perhaps this issue can be closed now, though a link to it somewhere would be nice.

I wonder if auto-generation of a bade-like image showing a translation progress summary would be possible?

**Comment by bl-ue** [comment]
>though a link to it somewhere would be nice.

I've added it to the readme in PR #5378.

>I wonder if auto-generation of a bade-like image showing a translation progress summary would be possible?

What do you think it would look like?  An overall percentage where each translation of each page counts?

**Comment by sbrl** [comment]
Yeah, and overall percentage. An SVG would probably be easiest, as then you could substitute values in using find / replace or something.

**Comment by bl-ue** [comment]
But it seems to me like it would update very slowly, and likely _decrease_ over time, because for **one** added English pages, there's a whole new row of ❌s (25 at the moment).

**Comment by Waples** [comment]
I have closed #5396 since 2+ years of no progress. Shall we close this issue aswel?

**Comment by kbdharun** [comment]
> I have closed #5396 since 2+ years of no progress. Shall we close this issue aswel?

We have https://lukwebsforge.github.io/tldri18n/ listed in README (also in some comments above) exactly for this. It lists the outdated, missing pages and the translation progress percentage for each platform, so I am closing this issue, as its primary objective has been achieved.

Also, if we are going with the SVG approach then we should opt for Graphs like in Weblate translation progress widgets i.e https://github.com/flathub/website#translations (feel free to check the backend code at https://github.com/WeblateOrg/weblate)
